### PR TITLE
docs changes; block names on controller, cbox read mode

### LINKED ISSUE #432: Update cbox docs to use ReadMode

### PR DESCRIPTION
**Pull request by steersbob**
Resolves #432﻿

### CODE REVIEW

File: docs/dev/reference/blocks_api.md [hunk 1]
```diff
@@ -138,9 +138,10 @@ Block data has two subsets: logged and stored data.
 
 Logged data consists of the fields that are published to the history service.
 This does not include string values, links, arrays, and some of the more persistent settings.
+You can't recreate a block's settings by writing its logged output.
```
Comment (elcojacobs) [inline docs/dev/reference/blocks_api.md]: recreate settings?
using mode logged with a write is an error and the API should throw an error

File: docs/dev/reference/blocks_api.md [hunk 2]
```diff
@@ -138,9 +138,10 @@ Block data has two subsets: logged and stored data.
...
 
-Stored data are the values persisted to EEPROM storage by the controller.
-Backups, for example, use stored data.
+Stored data consists of the fields that should be persisted to restore the block's settings after startup or from a backup.
```
Comment (elcojacobs) [inline docs/dev/reference/blocks_api.md]: at startup?

File: docs/dev/reference/cbox.md [hunk 3]
```diff
@@ -176,6 +183,93 @@ block data in `payload.content` is technically encoded four times:
...
+  DEFAULT = 0;
+  STORED = 1;
+  LOGGED = 2;
+}
+```
+
+Block create / read / write commands may specify the subset of fields they wish to receive in the response
+by using the `ReadMode mode` field in the request. \
+The mode signals the intent. It is up to the individual block to determine which fields match the mode.
+Blocks are not guarantee to return the same subset of fields for each read request with the same mode.
```
Comment (elcojacobs) [inline docs/dev/reference/cbox.md:212]: this sounds like mode doesn't deterministically determines which fields are included

File: docs/dev/reference/cbox.md [hunk 4]
```diff
@@ -176,6 +183,93 @@ block data in `payload.content` is technically encoded four times:
...
+
+message Payload {
+  ...
+  MaskMode maskMode = 6;
+  repeated MaskField maskFields = 7;
+}
+```
+
+Both requests and responses can use masks to indicate field presence.
+This feature is implemented to complement Protobuf's automatic omission of empty values,
```
Comment (elcojacobs) [inline docs/dev/reference/cbox.md]: zero values

File: docs/dev/reference/cbox.md [hunk 5]
```diff
@@ -215,7 +313,7 @@ enum Opcode {
 
 ### BLOCK_READ
 
-- Request payload: **Yes, blockId only**
+- Request payload: **Yes, block identity only**
```
Comment (elcojacobs) [inline docs/dev/reference/cbox.md]: identifier is probably the right word

File: docs/dev/reference/cbox.md (hunk 3 above)
Comment (steersbob) [reply docs/dev/reference/cbox.md:212]: It does not. That's the point.

**Pull request merged**